### PR TITLE
Refactoring combine notes

### PR DESCRIPTION
**Pull request by patnir**
## Summary

Moves spendPostTime to utils
Moves fetchNotes to utils
Moves TransactionTimer to its own file

This work is part of the work to add a timer to the transaction send command.

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and label it with `breaking-change-rpc` or `breaking-change-sdk`.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish-cli/src/commands/wallet/notes/combine.ts [hunk 1]
```diff
@@ -342,23 +191,6 @@ export class CombineNotesCommand extends IronfishCommand {
...
-    })
-
-    Assert.isNotNull(getBlockResponse.content.block.noteSize)
-
-    const config = await client.config.getConfig()
-
-    // Adding a buffer to avoid a mismatch between confirmations used to load notes and confirmations used when creating witnesses to spend them
-    return getBlockResponse.content.block.noteSize - (config.content.confirmations || 2)
-  }
-
```
Comment (patnir) [inline ironfish-cli/src/commands/wallet/notes/combine.ts:361]: Moved to a new file

File: ironfish-cli/src/commands/wallet/notes/combine.ts [hunk 2]
```diff
@@ -74,151 +68,6 @@ export class CombineNotesCommand extends IronfishCommand {
     }),
   }
 
-  private async getSpendPostTimeInMs(
```
Comment (patnir) [inline ironfish-cli/src/commands/wallet/notes/combine.ts:77]: All of the changes are just moving this to a new file

File: ironfish-cli/src/utils/transaction.ts [hunk 3]
```diff
@@ -14,70 +14,6 @@ import {
   UnsignedTransaction,
 } from '@ironfish/sdk'
 import { CliUx } from '@oclif/core'
-import { ProgressBar } from '../types'
-
-export class TransactionTimer {
-  private logger: Logger
```
Comment (patnir) [inline ironfish-cli/src/utils/transaction.ts:20]: moved to a transaction timer file

File: ironfish-cli/src/utils/timer.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  private estimateInMs: number
+  private timer: NodeJS.Timer | undefined
+
+  constructor(spendPostTime: number, raw: RawTransaction, logger?: Logger) {
+    this.logger = logger ?? createRootLogger()
+    this.estimateInMs = Math.max(Math.ceil(spendPostTime * raw.spends.length), 1000)
+  }
+
+  displayEstimate() {
+    this.logger.log(
```
Comment (NullSoldier) [inline ironfish-cli/src/utils/timer.ts:21]: you should just have a render function and the consumer should decide how to consume this. IE: commands should not use the logger, they should use `this.log` which is a base function in the Command.

Comment (patnir) [reply ironfish-cli/src/utils/timer.ts]: Thanks for the feedback, I was debating that too. Made the change

**Pull request closed without merging**